sql/analyzer/apply_join.go

```diff
@@ -394,7 +406,18 @@ func recurseSubqueryForOuterFilters(n sql.Node, a *Analyzer, scope *Scope) (sql.
 		if len(keepFilters) == 0 {
 			return f.Child, transform.NewTree, nil
 		}
-		ret := plan.NewFilter(expression.JoinAnd(keepFilters...), f.Child)
+		var keepInverted []sql.Expression
+		var keepNotInverted []sql.Expression
+		for _, h := range keepFilters {
+			if h.invert {
+				keepInverted = append(keepInverted, h.e)
```

Comment:
We should be inverting the expression here: `keepInverted = append(keepInverted, expression.NewNot(h.e))`

Reply:
This has been resolved, but the matching change (adding `expression.NewNot(h.e)`) has not been made.

Don't we need to invert the expressions at some point?

Reply:
We would need to interleave generic boolean simplification to do better here. So after making NOT(EXPR), trying to convert it into a conjugation of expressions.